### PR TITLE
add Shop Settings Update

### PR DESCRIPTION
**Pull request by AdamQuadmon**
new methods:

- update_shop_settings
- update_shop_domain
- update_shop_address

### CODE REVIEW

File: saleor_gql_loader/data_loader.py [hunk 1]
```diff
@@ -65,6 +65,117 @@ def __init__(self, auth_token, endpoint_url="http://localhost:8000/graphql/"):
...
+              }
+            }
+        """
+
+        response = graphql_request(
+            query, variables, self.headers, self.endpoint_url)
+
+        errors = response["data"]["shopSettingsUpdate"]["shopErrors"]
+        handle_errors(errors)
+
```
Comment (grll) [inline saleor_gql_loader/data_loader.py:119]: I think we should return an ID or something as a confirmation that the mutation worked

File: saleor_gql_loader/data_loader.py [hunk 2]
```diff
@@ -65,6 +65,117 @@ def __init__(self, auth_token, endpoint_url="http://localhost:8000/graphql/"):
...
+              }
+            }
+        """
+
+        response = graphql_request(
+            query, variables, self.headers, self.endpoint_url)
+
+        errors = response["data"]["shopDomainUpdate"]["shopErrors"]
+        handle_errors(errors)
+
```
Comment (grll) [inline saleor_gql_loader/data_loader.py:165]: simarly here we should also return an ID

File: saleor_gql_loader/data_loader.py [hunk 3]
```diff
@@ -65,6 +65,117 @@ def __init__(self, auth_token, endpoint_url="http://localhost:8000/graphql/"):
...
+              }
+            }
+        """
+
+        response = graphql_request(
+            query, variables, self.headers, self.endpoint_url)
+
+        errors = response["data"]["shopAddressUpdate"]["shopErrors"]
+        handle_errors(errors)
+
```
Comment (grll) [inline saleor_gql_loader/data_loader.py:225]: similarly here we should also return and ID

File: saleor_gql_loader/data_loader.py [hunk 4]
```diff
@@ -65,6 +65,117 @@ def __init__(self, auth_token, endpoint_url="http://localhost:8000/graphql/"):
         self.headers = {"Authorization": "Bearer {}".format(auth_token)}
         self.endpoint_url = endpoint_url
 
+    def update_shop_settings(self, **kwargs):
```
Comment (grll) [inline saleor_gql_loader/data_loader.py:68]: the id of the shop is required ? If so we should make it mandatory as a separate argument of the function

**Review by grll — CHANGES_REQUESTED** [review]
Great job thanks! Just reviewed a few details

File: saleor_gql_loader/data_loader.py (hunk 1 above)
Comment (AdamQuadmon) [reply saleor_gql_loader/data_loader.py:119]: right, but shop doesn't have ID if I'm not wrong
we can return the shop object with updated fields

File: saleor_gql_loader/data_loader.py (hunk 4 above)
Comment (AdamQuadmon) [reply saleor_gql_loader/data_loader.py:68]: Nope, there is just one shop supported in saleor for now, no ID required

I also checked for mandatory fields or default suggestions but didn't find any

File: saleor_gql_loader/data_loader.py (hunk 2 above)
Comment (AdamQuadmon) [reply saleor_gql_loader/data_loader.py:165]: About the ShopDomain it's a bit ambiguous

The queried Shop object has a Domain type with the following return values:
```
domain {
  host
  sslEnabled
  url
}
```

but the SiteDomainInput is the following:
```
input SiteDomainInput {
  domain: String
  name: String
}
```

File: saleor_gql_loader/data_loader.py (hunk 1 above)
Comment (grll) [reply saleor_gql_loader/data_loader.py:119]: yes perfect

File: saleor_gql_loader/data_loader.py (hunk 2 above)
Comment (grll) [reply saleor_gql_loader/data_loader.py:165]: as long as it returns something related to the updated / created object I think it's fine

**Review by grll — APPROVED** [review]

File: saleor_gql_loader/data_loader.py (hunk 4 above)
Comment (grll) [reply saleor_gql_loader/data_loader.py:68]: ok perfect then

**Pull request merged**